rviz_default_plugins/src/rviz_default_plugins/tools/pose_estimate/initial_pose_tool.cpp

```diff
@@ -74,7 +74,7 @@ void InitialPoseTool::onPoseSet(double x, double y, double theta)
 
   geometry_msgs::msg::PoseWithCovarianceStamped pose;
   pose.header.frame_id = fixed_frame;
-  pose.header.stamp = rclcpp::Clock().now();
+  pose.header.stamp = context_->getRosNodeAbstraction().lock()->get_raw_node()->now();
```

Comment:
The context also has a clock (i.e. `context_->getClock()->now()`), why do you not use that one?

Reply:
I gave that a shot, and while it produces the time, it always seems to produce wall time regardless of the `use_sim_time` parameter - the same behavior as `rclcpp::Clock().now()`.

It may be a better approach to track down why the parameter value isn't respected - I'm not sure.

Reply:
I'm guessing that the context has the clock but doesn't use the node to attach a TimeSource to it so it never gets updated from the parameter or via the clock topic.

Reply:
That makes sense. When porting RViz, the time API was still under construction and the necessity of attaching the time source escaped us.

Here is where the clock is constructed: https://github.com/ros2/rviz/blob/a20152c290739c4cf2ef5af06a7e498a744162ac/rviz_common/src/rviz_common/visualization_frame.cpp#L328

At that point, the rviz node already exists (it's used to call the constructor of `VisualizationManager`, which is actually the `DisplayContext` in the next line). So I'd suggest attaching the TimeSource to the node at that point. 

The cleanest way to do this would probably be to add a method to the `ros_integration` so that the raw node doesn't need to be extracted.

Reply:
If you have the node at the point that the clock is initialized in the context, you could just store a pointer to the node's clock, and continue using it accessing it through the context's pointer to the clock as proposed in the first comment here.

Reply:
Rebased and updated to take that approach.